frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsAndSoil.tsx

```diff
@@ -17,117 +17,140 @@ import CustomTextArea from '@Commons/Input/TextArea/CustomTextArea';
-  StyledFarmInfo,
-  StyledTextAreaContainer,
-  StyledButtonGroupContainer,
-  StyledListContainer,
-  StyledListItem,
-} from './FieldsAndSoil.style';
-
-const initialFieldValues = initialFarmDetails.Fields[0];
+import FieldsButtonComponent from './FieldsButtonComponent';
+import FieldsListComponent from './FieldsListComponent';
+import { StyledFarmInfo, StyledTextAreaContainer } from './FieldsAndSoil.style';
+import { StyledButtonGroupContainer } from './FieldsButtonComponent.styles';
 
 const FieldsAndSoilComponent: React.FC<InputModuleProps> = ({ farmDetails, updateFarmDetails }) => {
-  const [fieldsInfo, setFieldsInfo] = useState(farmDetails);
+  // Builds field info inside the field form module.
+  const [, setFieldsInfo] = useState(farmDetails);
+  const [fieldIndex, setFieldIndex] = useState(0);
+  const [initialFieldValues, setInitialFieldValues] = useState(
+    initialFarmDetails.Fields[fieldIndex],
+  );
+  // Only triggered once, it would show list and persists.
+  const [isSubmitted, setSubmitted] = useState<boolean>(false);
+  // Would trigger when new field button is clicked.
+  const [fieldAdd, setFieldAdd] = useState<boolean>(false);
 
   const validationSchema = Yup.object().shape({
     FieldName: Yup.string().max(24).required('Required'),
-    Area: Yup.number().min(1).max(100).required('Required'),
-    Comments: Yup.string().max(200),
+    Area: Yup.number()
+      .min(1, 'Area should be higher than 1')
+      .max(100, 'Area should be lower than 100')
+      .required('Required'),
+    Comments: Yup.string().max(200, 'Comments should be lower than 200'),
   });
-
-  const onSubmit = (
-    values: FieldDetailInterface,
-    { setSubmitting }: { setSubmitting: (isSubmitting: boolean) => void },
-  ): void => {
+  /**
+   *
+   * @param values : It's of type FieldDetailInterface, it calls, FieldName, Area, and Comments
+   * @desc         User data is inputted and then pushed into FarmDetailsInterface,
+   *               initial values for formik is then set to empty values, state hooks gets triggered
+   * @author       @Kcaparas
+   */
+  const addFieldData = (values: FieldDetailInterface): void => {
     setTimeout(() => {
-      const newFarmDetails: FarmDetailsInterface = { ...farmDetails };
-      const newField: FieldDetailInterface = initialFieldValues;
-      newField.FieldName = values.FieldName;
-      newField.Area = values.Area;
-      newField.Comments = values.Comments;
-      newFarmDetails.Fields.push(newField);
-      updateFarmDetails(newFarmDetails);
-      setSubmitting(false);
+      const farmInfo: FarmDetailsInterface = { ...farmDetails };
+      farmInfo.Fields.push({
+        id: fieldIndex,
+        FieldName: values.FieldName,
+        Area: values.Area,
+        Comments: values.Comments,
+      });
+      const newInitialFieldValues = {
+        id: fieldIndex,
+        FieldName: '',
+        Area: 0,
+        Comments: '',
+      };
+      setInitialFieldValues(newInitialFieldValues);
```

Comment:
Is it possible to use `setInitialFieldVakues(InitialFarmDetails.Fields[0])`?
Could simplify this a bit :smile:

Reply:
I guess, we can initialize the id to 1 anyways. I'll check it out.